### PR TITLE
Fix index out of bound exception when componsating

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
This PR fixes the exception that is thrown when compensation action is returning to caller.

## Approach
Compensation block is modeled as a lambda function and when executing it we need to provide the return registry index even though the lambda is a void function.

An index to empty slot from the callers `workerLocal` is passed to the `compensate` instruction.
So that it doesn't overwrite live workerLocals.

### CODE REVIEW

**Comment by anupama-pathirage** [comment]
@rdhananjaya Can you please add a test case for this

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java [hunk 1]
```diff
@@ -1384,8 +1384,8 @@ public void visit(BLangCompensate compensate) {
         Stack children = childScopesMap.get(compensate.scopeName.getValue());
 
         int i = 0;
-        //scopeName, child count, children
-        Operand[] operands = new Operand[2 + children.size()];
+        //scopeName, child count, children, NIL return
```
Comment (Shan1024) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: Better to have a space after `//` and end the comment with a full stop.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java [hunk 2]
```diff
@@ -1396,6 +1396,8 @@ public void visit(BLangCompensate compensate) {
             int childNameCP = currentPkgInfo.addCPEntry(new UTF8CPEntry(childName));
             operands[i++] = getOperand(childNameCP);
         }
+        // compensation block is modeled as an anonymous function, which require a return reg
```
Comment (Shan1024) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: Better to start with an uppercase letter and end the comment with a full stop.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/Mnemonics.java [hunk 3]
```diff
@@ -242,6 +242,11 @@
         mnemonics[InstructionCodes.XML2S] = "xml2s";
         mnemonics[InstructionCodes.LOCK] = "lock";
         mnemonics[InstructionCodes.UNLOCK] = "unlock";
+
+        mnemonics[InstructionCodes.SCOPE_END] = "scope_end";
+        mnemonics[InstructionCodes.COMPENSATE] = "compensate";
+        mnemonics[InstructionCodes.LOOP_COMPENSATE] = "loop_compensate";
+
```
Comment (Shan1024) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/Mnemonics.java]: Extra newline?

**Pull request merged**